### ISSUE TITLE
[TCC] Evaluation gives very high loss value

### ISSUE THREAD

**Issue by yongwanlim** [google-research/google-research#699]
Hi @debidatta

Once training is completed, I usually see that I get the training loss of around 0.005 with 60000 iterations for my settings.
But for evaluation when I run `python -m tcc.evaluate --alsologtostderr --logdir $MODEL_DIR`, I get a very high training loss like ~1.000. And I got a very small value for the kendalls tau task like 0.02. Though alignment looks working quite well. 

I am wondering what could cause this discrepancy in the training loss between training and evaluation. Any feedback would be appreciated. Thank you!

**Comment by debidatta** [comment]
Hi @yongwanlim thanks for reaching out. 

Can you check the logs of the evaluate script to see if the checkpoint is getting loaded properly? Another thing to check would be if the function calls in eval script are being run in eval mode (training=False in Keras model calls). This used to happen implicitly before due to tf.keras.backend.set_learning_phase(0) but this feature was supposed to be deprecated in newer TF versions.

**Comment by yongwanlim** [comment]
Hi @debidatta, 

Thanks for the suggestions. 

I think I resolved the issue. I had an issue in the following line: 
`restore_ckpt(logdir=CONFIG.LOGDIR, optimizer=optimizer, **algo.model)` in `evaluate_once` function.

`CONFIG.LOGDIR` is being updated in `evaluate()` by assigning `CONFIG.LOGDIR = FLAGS.logdif` but in `evaluate_once()` when loading the checkpoint, `CONFIG.LOGDIR` was not actually `FLAGS.logdir` so I happened to refer to a wrong `logdir`.  I had to add `CONFIG.LOGDIR = FLAGS.logdir` again at the beginning of `evaluate_once` function.

Regarding your second suggestion, I have questions. 

Shouldn't the following update make the base_model be operated in eval mode?     
```
training = (tf.keras.backend.learning_phase() and CONFIG.MODEL.TRAIN_BASE != 'frozen')
x = self.base_model(images, training=training)
```

Should I additionally modify ConvEmbedder for the eval mode? For example, turning off fc_dropout and/or inputting training=false for batch normalization? 

Any feedback would be appreciated. Thank you!

**Comment by debidatta** [comment]
Yes you are right. That is the reason the eval is off.

I rechecked the second suggestion and you don't need to change anything if you are using the version of TF mentioned in requirements.txt.

For newer versions, tf.keras.backend.learning_phase() will be deprecated and you need to pass training=True/False explicity. The part you pasted already does this, so that's ok. The part that will need changing in the future is https://github.com/google-research/google-research/blob/master/tcc/utils.py#L302 and https://github.com/google-research/google-research/blob/master/tcc/utils.py#L396 where training=False needs to be passed. 

Let me know if eval works for you now.
